### ISSUE TITLE
Target function returns NaN

### ISSUE THREAD

**Issue by bonh** [python-adaptive/adaptive#435]
I have a complex target function that involves a nonlinear optimization under the hood. The optimizer might fail. What should the target function return to indicate to `adaptive` that this region of the parameter space is no good? `np.nan` ?

**Comment by basnijholt** [comment]
You probably need a custom loss function. Where if a NaN is returned, you set the region to 0 loss.

Which learner are you using?

**Comment by bonh** [comment]
Okay, sounds good, I'll try that. 

I'm using `LearnerND` for $R^3 \rightarrow R^1$.

**Comment by basnijholt** [comment]
OK, then try something like:
```python
def custom_loss(simplex, values, value_scale):
    if any(np.isnan(v) for v in values):
        return 0
    return adaptive.learner.learnerND.default_loss(simplex, values, value_scale)

learner = adaptive.LearnerND(..., loss_per_simplex=custom_loss)
```

Note that I didn't test this 😅

**Comment by basnijholt** [comment]
Maybe you just want to make it less likely that something in that area gets chosen. You can do that with
```
def custom_loss(simplex, values, value_scale):
    loss = adaptive.learner.learnerND.default_loss(simplex, values, value_scale)
    if any(np.isnan(v) for v in values):
        return loss / 10
    return loss

learner = adaptive.LearnerND(..., loss_per_simplex=custom_loss)

```

**Comment by bonh** [comment]
That seems to work - thanks!
